src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs

```diff
@@ -686,6 +712,21 @@ private void ProcessApp1Marker(BufferedReadStream stream, int remaining)
                     this.ExtendProfile(ref this.exifData, profile.AsSpan(Exif00).ToArray());
                 }
             }
+
+            if (ProfileResolver.IsProfile(profile, ProfileResolver.XmpMarker))
+            {
+                this.isXmp = true;
+                if (this.xmpData is null)
+                {
+                    // The first 29 bytes will be skipped, because this is Jpeg specific
+                    this.xmpData = profile.AsSpan(XmpNsLength).ToArray();
```

Comment:
If this can grow large, we should request `profile`  from `MemoryAllocator` so it's pooled, to at least avoid double GC allocation. (Once with `new byte[remaining]` once with `.ToArray()`)

Reply:
I have reduced the allocations, but MemoryAllocator is not an option for profiles, see previous comment from @JimBobSquarePants .